scripts/install_apt_deps.sh

```diff
@@ -1,21 +1,301 @@
+# package groups
+pkg_dep_list=$(pkg_dep_list_from "${pkg_dep_groups_prereq}")
+echo "Installing prerequisite packages: ${pkg_dep_list}"
+
+# Refresh package list and grab prerequisite packages needed for package
+# and repository management within this script
+sudo apt-get update
+sudo apt-get install -q -y ${pkg_dep_list}
+
+# Add repositories needed for drivers and miscellaneous dependencies (python)
 sudo add-apt-repository -u -y ppa:graphics-drivers/ppa
 sudo add-apt-repository -u -y ppa:deadsnakes/ppa
-sudo apt-get install -y \
-	 git clang-10 make cmake libc++-dev libc++abi-dev unzip \
-	 libsqlite3-dev libeigen3-dev libboost-all-dev libatlas-base-dev libsuitesparse-dev libblas-dev libtbb-dev \
-	 glslang-tools libsdl2-dev libglu1-mesa-dev mesa-common-dev freeglut3-dev libglew-dev glew-utils libglfw3-dev \
-	 libusb-dev libusb-1.0 libudev-dev libv4l-dev libhidapi-dev \
-	 build-essential libx11-xcb-dev libxcb-glx0-dev libxcb-randr0-dev libxrandr-dev libxkbcommon-dev libwayland-dev \
-	 libepoxy-dev libdrm-dev libgbm-dev libx11-dev libvirglrenderer-dev libpulse-dev \
-	 libgtest-dev pkg-config libgtk2.0-dev wget xvfb librealsense2-dkms librealsense2-utils \
-     libjpeg-dev libpng-dev libtiff-dev libvtk6-dev libgtk-3-dev libparmetis-dev gfortran \
-	 libgflags-dev libgoogle-glog-dev \
+
+# Add Kitware repository (for third party Ubuntu dependencies)
+key_srv_url_kitware="https://apt.kitware.com/keys/kitware-archive-latest.asc"
+repo_url_kitware="https://apt.kitware.com/ubuntu"
+add_repo "${key_srv_url_kitware}" "${repo_url_kitware}" "${VERSION_CODENAME} main"
+
+# If prompted, add Docker repository (for local CI/CD debugging)
+if [ "${use_docker}" == "yes" ]; then
+    key_srv_url_docker="https://download.docker.com/linux/ubuntu/gpg"
+    repo_url_docker="https://download.docker.com/linux/ubuntu"
+    add_repo "${key_srv_url_docker}" "${repo_url_docker}" "${VERSION_CODENAME} stable"
+fi
+
+# If supported, add the gpg keys and repository for Intel RealSense
+if [ "${use_realsense}" == "yes" ]; then
+    key_srv_url_list_realsense="keys.gnupg.net hkp://keyserver.ubuntu.com:80"
+    repo_url_realsense="http://realsense-hw-public.s3.amazonaws.com/Debian/apt-repo"
+    key_id_realsense="F6E65AC044F831AC80A06380C8B3A55A6F3EFCDE"
+    add_repo "${key_srv_url_list_realsense}" "${repo_url_realsense}" "${VERSION_CODENAME} main" "${key_id_realsense}"
+fi
+
+# If supported, add the keys and repository for CUDA (for GPU plugin support)
+if [ "${use_cuda}" == "yes" ]; then
+    distro_cuda="ubuntu1804"
+    arch_cuda="x86_64"
+    repo_url_cuda="https://developer.download.nvidia.com/compute/cuda/repos/${distro_cuda}/${arch_cuda}"
+    key_srv_url_cuda="${repo_url_cuda}/7fa2af80.pub"
+    add_repo "${key_srv_url_cuda}" "${repo_url_cuda}" "/"
+
+    path_cmd_cuda='export PATH=/usr/local/cuda-11.1/bin${PATH:+:${PATH}}'
+    lib64_cmd_cuda='export LD_LIBRARY_PATH=/usr/local/cuda-11.1/lib64${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}'
```

Comment:
I agree this is an issue. Previously we relied on the user to go through the documentation for their system. This path setup is part of the [post installation actions](https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html#environment-setup) shared for all configurations. Our scripts runs through most of the current process for an Ubuntu system for CUDA 11, which makes it easier to get started from a new install. Do you think a message stating that this is only for Ubuntu and CUDA 11?

Reply:
Yeah, I think we should add such a message, and also make it bold and clear (all caps or whatever) that the user should *only* do this if they are sure there won't be issues -- out of everything we install, NVIDIA drivers and CUDA have the highest likelihood of bricking someone's machine, and we don't want to be liable for that (our license says this, but who really reads those).

Reply:
How do lines `scripts/install_apt_deps.sh:231-236` look to you?

Reply:
They look good. But I was thinking of `install_deps.sh:93`. Instead of a simple query like for the other installs, make the CUDA query big, bold, and red (jk). 😛